File: backend/src/main/java/nf/fr/k49/collektor/model/Collektion.java
package nf.fr.k49.collektor.model;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public record Collektion(
    String id,
    String label,
    String imgUrl,
    List<CollektionItemField> customFields
) {
    public Collektion setId(String newId) {
        return new Collektion(newId, this.label(), this.imgUrl(), this.customFields());
    }
};

File: frontend/src/app/model/Collektion.ts
export interface Collektion {
    id: string;
    label: string;
    imgUrl?: string;
    customFields: CollektionItemField[];
}

export interface CollektionItemField {
    key: string;
    label: string;
    type: CollektionItemFieldType;
}

export type CollektionItemFieldType = 'TEXT'  | 'BOOLEAN' | 'INTEGER' | 'DATE';
export const CollektionItemFieldTypes: CollektionItemFieldType[] = [ 'TEXT', 'BOOLEAN', 'INTEGER', 'DATE' ];